### PR TITLE
wait for instance status only after create/update

### PR DESCRIPTION
**Pull request by ScottSuarez**
closes https://github.com/hashicorp/terraform-provider-google/issues/9631

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
compute: changed `wait_for_instances` in `google_compute_instance_group_manager` and `google_compute_region_instance_group_manager` to no longer block plan / refresh, waiting on managed instance statuses during apply instead
```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb [hunk 1]
```diff
@@ -626,27 +633,6 @@ func resourceComputeInstanceGroupManagerRead(d *schema.ResourceData, meta interf
 		return fmt.Errorf("Error setting status in state: %s", err.Error())
 	}
 
-	// If unset in state set to default value
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb:629]: Don't think we want to change this bit.

Comment (ScottSuarez) [reply mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb:629]: reverted

**Comment by ScottSuarez** [comment]
looks like everything passed @rileykarson

File: mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb [hunk 2]
```diff
@@ -758,6 +744,13 @@ func resourceComputeInstanceGroupManagerUpdate(d *schema.ResourceData, meta inte
 
 	d.Partial(false)
 
+	if d.Get("wait_for_instances").(bool) {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb:747]: To be equivalent to the old check in Read shouldn't we wait pre-update and pre-delete as well?

Comment (ScottSuarez) [reply mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb:755]: that tracks for me

File: mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb [hunk 3]
```diff
@@ -25,8 +25,8 @@ func resourceComputeInstanceGroupManager() *schema.Resource {
 			State: resourceInstanceGroupManagerStateImporter,
 		},
 		Timeouts: &schema.ResourceTimeout{
-			Create: schema.DefaultTimeout(5 * time.Minute),
-			Update: schema.DefaultTimeout(5 * time.Minute),
+			Create: schema.DefaultTimeout(15 * time.Minute),
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb:28]: Make sure to fix the docs too! https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance_group_manager#timeouts

File: mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb [hunk 4]
```diff
@@ -815,10 +824,32 @@ func resourceComputeInstanceGroupManagerDelete(d *schema.ResourceData, meta inte
 		err = computeOperationWaitTime(config, op, project, "Deleting InstanceGroupManager", userAgent, d.Timeout(schema.TimeoutDelete))
 	}
 
+	// If unset in state set to default value
+	if d.Get("wait_for_instances_status").(string) == "" {
+		if err := d.Set("wait_for_instances_status", "STABLE"); err != nil {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb:829]: This block should be in `Read` and not `Delete` right?

File: mmv1/third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb [hunk 5]
```diff
@@ -24,8 +24,8 @@ func resourceComputeRegionInstanceGroupManager() *schema.Resource {
 			State: resourceRegionInstanceGroupManagerStateImporter,
 		},
 		Timeouts: &schema.ResourceTimeout{
-			Create: schema.DefaultTimeout(5 * time.Minute),
-			Update: schema.DefaultTimeout(5 * time.Minute),
+			Create: schema.DefaultTimeout(15 * time.Minute),
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb:27]: Same for these timeouts!

**Review by rileykarson — COMMENTED** [review]
Your changelog entry is out of date now, right? I'd suggest something user-focused eg:

```
compute: changed `wait_for_instances` in `google_compute_instance_group_manager` and `google_compute_region_instance_group_manager` to no longer block plan / refresh, waiting on managed instance statuses during apply instead
```

File: mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb (hunk 4 above)
Comment (ScottSuarez) [reply mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb]: yeah this is my bad, I'll take things a little slower next time

File: mmv1/third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb (hunk 5 above)
Comment (ScottSuarez) [reply mmv1/third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb:27]: done!

File: mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb (hunk 3 above)
Comment (ScottSuarez) [reply mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb:28]: done !

**Review by rileykarson — APPROVED** [review]

**Pull request merged**